core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -73,14 +77,14 @@ private[log] class ProducerAppendInfo(val producerId: Long, initialEntry: Produc
   private var coordinatorEpoch = initialEntry.coordinatorEpoch
   private val transactions = ListBuffer.empty[TxnMetadata]
 
-  def this(producerId: Long, initialEntry: Option[ProducerIdEntry], loadingFromLog: Boolean) =
-    this(producerId, initialEntry.getOrElse(ProducerIdEntry.Empty), loadingFromLog)
+  def this(producerId: Long, initialEntry: Option[ProducerIdEntry], validateSequenceNumbers: Boolean, loadingFromLog: Boolean) =
+    this(producerId, initialEntry.getOrElse(ProducerIdEntry.Empty), validateSequenceNumbers, loadingFromLog)
```

Comment:
I found it a bit confusing to have this secondary constructor. What about something like:

https://github.com/ijuma/kafka/commit/776ba5fd64c3e1062d2905e3d6314675f8174e2c